### PR TITLE
Ability to delete account as a social user { Continued in #8796 }

### LINKED ISSUE #7417: can't delete account if you use Facebook or Google authentication
The delete account button now requires that you enter your password.

Players who use only Facebook authentication don't have a password. The delete feature needs to have an exception for them where they need to enter only "DELETE" (localised) as previously.

### PR DESCRIPTION
**Pull request by rae-ralston**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #7417

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

Updated so that users without a local log in information can still delete their account by typing DELETE instead of a password. When the settings page is loaded if `user.auth.local.email` is present then the modal requiring the password pops up. If it's not present then it's assumed that the person is logged in via social and they're given the DELETE modal.

social delete modal

The user route handles verification, and checks if both `user.auth.local.email` & `user.auth.local.hashed_password` are true and if so, uses the previous written flow to delete a user. 

If `user.auth.google.id` or `user.auth.facebook.id` are true then the logic checks that the user wrote DELETE (case sensitive).

#### File Changes
- **front.json & settings.json**: added phrases that match the updated UI elements.
- **website/server/controllers/api-v3/user.js**: added an `else/if` statement that checks if the user has a local or social log in and check the respective expected input entry.
- **website/views/options/settings.jade**: added another button and if logic to both so the correct button will display depending on whether or not the user has local auth.
- **website/views/shared/modals/settings.jade**: added the social modal content.

I'd love it if someone could give me some reflections on the best things to test the validity of a local user account. Can someone get an email into local if they don't have a local registration? I couldn't get a real idea of what social user data is stored in `user.auth.google/facebook`, is looking for `.id` specific enough?

Looking forward to reflections 💃 

big ups to @pllearns for writing the testing and @deonna for help and support <3 

[phil's testing PR](https://github.com/HabitRPG/habitica/pull/8429)

----
rachel-ftw's UUID: 97f4775f-0e99-47cd-8002-63a062972bf6

### CODE REVIEW

**Comment by rae-ralston** [comment]
Added image to initial comment: 
social delete modal

and referenced [phil's testing PR](https://github.com/HabitRPG/habitica/pull/8429)

**Comment by Alys** [comment]
@rachel-ftw If you haven't already, see my comment here and the one after it: https://github.com/HabitRPG/habitica/issues/7417#issuecomment-274313672
That might answer some of your questions. Tell us if you have others.

File: website/server/controllers/api-v3/user.js [hunk 1]
```diff
@@ -191,17 +191,24 @@ api.deleteUser = {
...
+        password: {
+          notEmpty: {errorMessage: res.t('missingPassword')},
+        },
+      });
+      let oldPassword = passwordUtils.encrypt(req.body.password, user.auth.local.salt);
+      if (oldPassword !== user.auth.local.hashed_password) {
+        throw new NotAuthorized(res.t('wrongPassword'))
+      }
+    } else if (user.auth.facebook.id || user.auth.google.id) {
+      if(req.body.password !== 'DELETE') {
```
Comment (TheHollidayInn) [inline website/server/controllers/api-v3/user.js]: I would combine this with the above `else if`

Comment (TheHollidayInn) [reply website/server/controllers/api-v3/user.js]: Also, let's move DELETE up to be a const.

**Review by TheHollidayInn — COMMENTED** [review]
Looks good. I believe I saw test in a separate PR. Are you two working together on this?

File: website/server/controllers/api-v3/user.js (hunk 1 above)
Comment (Alys) [reply website/server/controllers/api-v3/user.js]: "DELETE" is intended to be translatable so a constant wouldn't work.

Comment (rae-ralston) [reply website/server/controllers/api-v3/user.js]: fixed the `else if`

**Comment by rae-ralston** [comment]
@TheHollidayInn Yah, @pllearns and I were working together on it. ❤️

**Comment by TheHollidayInn** [comment]
Awesome, if you can fix the linting issues and merge conflicts, I'll check the other PR is well.

File: website/server/controllers/api-v3/user.js [hunk 2]
```diff
@@ -191,17 +191,22 @@ api.deleteUser = {
...
-      },
-    });
-
     let validationErrors = req.validationErrors();
     if (validationErrors) throw validationErrors;
 
-    let oldPassword = passwordUtils.encrypt(req.body.password, user.auth.local.salt);
-    if (oldPassword !== user.auth.local.hashed_password) throw new NotAuthorized(res.t('wrongPassword'));
+    if(user.auth.local.hashed_password && user.auth.local.email) {
+      req.checkBody({
```
Comment (paglias) [inline website/server/controllers/api-v3/user.js:198]: req.checkBody doesn't do anything unless you call req.validationErrors() and then throw the error if it exists, like it's done a few lines above

**Review by paglias — CHANGES_REQUESTED** [review]

**Comment by SabreCat** [comment]
Continuing this in #8796. @rachel-ftw, if you'd like to resume work on this yourself, you need only say the word!

**Pull request closed without merging**